src/main/java/org/opensearch/security/transport/SecurityRequestHandler.java

```diff
@@ -326,6 +327,11 @@ protected void addAdditionalContextValues(final String action, final TransportRe
             }
         }
 
+        String extensionUniqueId = getThreadContext().getHeader("extension_unique_id");
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS) && extensionUniqueId != null) {
```

Comment:
**Can this check be stronger?**

Absolutely, this check is basic at the moment and only checks for the presence of the header. A simple improvement on this would be to check not only the existence of the header, but also that it is a value contained in `extensions/extensions.yml`.

There should also be a check similar to `nodes_dn` that verifies that the principal extracted from the certificate of the extension matches a list of known `extensions_dns` that would be placed in the appropriate area of the `extensions/extensions.yml` configuration for the extension.

Reply:
I modified this to check for the ID in the extensions registry.

Reply:
Is there a way we could remove the `FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)` check?  It seems redundant as the extension_unique_id value won't be set and the ExtensionsManager won't find any matching extensions

Reply:
@cwperks Seems like there is a principal you'd like to see in how some features work that is at odds with our other practices.  If you were to ask me what I think your principal is I'd state it as "Functionality for extensions should be behind a feature flag(s) to prevent issues on clusters that do not have extensions enabled". Does this seem correct, did you want to expand on this?

This is a counterpoint to "Reduce coupling whenever possible".  When we make a call to a static function call to determine global state the responsibility of the code has been broadly increased, making maintenance more difficult.

I think this dialog represents different principals we'd like to see prioritized in the code, it might documenting and discussing as a tenant, what do you think?

Reply:
@peternied Yes that's correct. In this particular case its new functionality in the security plugin that would directly let an extension handshake with an opensearch node and also submit transport requests to initialize the extension like registering its REST handlers with OpenSearch - all of this is specific to extensions and is not permitted for anything other than an extension. In this case, I can't see any abstraction to add here because it is functionality for extensions and since other extension functionality is gated with a feature flag I think its appropriate to include here too.

I don't see this as coupling necessarily except that there are 3 places that now rely on the same threadcontext header and I'm not sure how to do that otherwise. 

Cross cluster scenarios are simpler because its the security plugin that populates the threadcontext header so the security plugin is the only codebase that needs to be aware of what that header is. I can't see a better way here currently.

Reply:
@DarshitChanpura @peternied I removed the feature flag check. There currently is no great way to test this change. I think the best way to automate a check for this change will be an integration test that verifies that an extension can be initialized when connecting with an opensearch node with the security plugin installed. There is an open issue on the sdk repo for [integration tests](https://github.com/opensearch-project/opensearch-sdk-java/issues/715) that will eventually validate this change.